srcmedia/ts/lib/form.ts

```diff
@@ -0,0 +1,59 @@
+import { Subject } from 'rxjs'
+
+import { Reactive, Component } from './common'
+
+interface RxFormState {
+    // eventually this could include generic things like validity
+}
+
+class RxForm extends Component implements Reactive<RxFormState> {
+    element: HTMLFormElement
+    state: Subject<RxFormState>
+    target: string
+
+    constructor(element: HTMLFormElement) {
+        super(element)
+        this.state = new Subject()
+        // FIXME naively assuming that the form targets the current path
+        this.target = window.location.pathname
```

Comment:
could check for form action attribute and default to current url if not set

Reply:
I tried doing this but setting `action=` manually causes the form to fail when js is turned off, and I can't figure out how to get the embedded `action` property out of the Django form, since it seems like Django just "knows" it?

Reply:
although I guess there's no harm in checking, just seems like django forms generally don't have `action=`

Reply:
We don't usually bother putting one when we're submitting to the same url that the form is on.  It's not a django-specific thing, just taking advantage of default browser behavior to submit to the current url if no action is set.

Thinking about it more, I'm not sure a reactive search would make sense for a form that submits to a different url... 

I'm ok with whichever way you want to handle it; fine if you just revise the comment to say that's current practice for django forms we want to be reactive.